### PR TITLE
Update dev mode docs

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR updates the docs to describe the latest dev mode usage 

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/dev_mode.md [hunk 1]
```diff
@@ -8,33 +8,19 @@ Arbitrary agent services built with the {{open_autonomy}} framework can be run w
...
+$ autonomy build-images --dependencies
+$ autonomy build-images --dev
 ```
-This builds the latest docker images and launches docker compose. Wait for environment setup to be completed inside your docker containers, it may take some time. Also don't make any changes on either open-aea or open-autonomy repository while environment is being set up it may cause some unexpected errors.
 
-Then run:
-```bash
-make run-deploy
-```
+This will create a deployment with hot reload enabled for agents. You can run it using the same methods as normal deployments. Use `autonomy deploy run` or `docker-compose up --force-recreate` to start the deployment and enjoy buiding the services.
```
Comment (dvilelaf) [inline docs/dev_mode.md:21]: Here you mention hot reload, below hot reload has been changed to dev mode. Shouldnt we update here as well?

File: docs/dev_mode.md [hunk 2]
```diff
@@ -46,18 +32,10 @@ Once the agents are running, you can make changes to the agent's packages as wel
 The trigger is caused by any python file closing in either `open-autonomy/packages` or `open-aea/` directory. So even if you haven't made any change and still want to restart the agent, just open any python file press `ctrl+s` or save it from file menu and it will trigger the restart.
 
 
-## Quick start for oracle
-
-We have a single make target for the oracle app, just run the following command and it will build and run the oracle.
-```bash
-make run-oracle-dev
```
Comment (DavidMinarsch) [inline docs/dev_mode.md:53]: Somewhat unrelated to the PR, but I spotted it here. Please remove the make commands in these files too:
docs/application_deployment.md
docs/autonomy.md
docs/benchmarks.md
docs/counter_example.md
docs/hot_reload.md
docs/using_stack_deployment.md

In cases where the commands are needed, a sample makefile with the commands should be added to the docs. Basically, it must be possible to use it without pulling the repo.

File: docs/dev_mode.md [hunk 3]
```diff
@@ -8,33 +8,19 @@ Arbitrary agent services built with the {{open_autonomy}} framework can be run w
...
-*Tip: You can also store above variables in your .bashrc file*
-
-
-### Prepare images & run
-
-Execute the following:
 ```bash
-make build-deploy
+# build dev deployment using
+$ autonomy deploy build --dev --packages-dir PATH_TO_PACKAGES_DIR --open-aea-dir PATH_TO_LOCAL_OPEN_AEA_REPO --open-autonomy-dir PATH_TO_LOCAL_OPEN_AUTONOMY_DIR
```
Comment (dvilelaf) [inline docs/dev_mode.md:15]: So we need 3 different directories? Does the user need to clone both open-aea and open-autonomy then?

File: docs/dev_mode.md (hunk 1 above)
Comment (angrybayblade) [reply docs/dev_mode.md:21]: Hot reload is just a feature which can be used in the dev mode. The dev mode is a setup which provides various features (Right now it's hot reload, storing data for replay and benchmarking).

File: docs/dev_mode.md (hunk 3 above)
Comment (angrybayblade) [reply docs/dev_mode.md:15]: Yeah, if someone wants to use the dev mode they need to clone the open-aea and open-autonomy repo but not the packages dir. The packages dir is supposed to point at the local repo where they store their dev packages.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**